### ISSUE TITLE
Handle more file closing with "with"

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8574]
|[<img alt="scop's avatar" src="https://avatars.githubusercontent.com/u/109152?s=50" width="50" height="50">](https://github.com/scop)| @scop reported|
|-|-|
|Trac ID|trac#8574|
|Type|enhancement|
|Created|2016-07-09 03:57:38Z|
|Branch|https://github.com/twisted/twisted/tree/8574-scop-opens|

See:
https://github.com/twisted/twisted/pull/299

<details><summary>Searchable metadata</summary>

```
trac-id__8574 8574
type__enhancement enhancement
reporter__scop scop
priority__normal normal
milestone__PyPy_support PyPy-support
branch__8574_scop_opens 8574-scop-opens
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1468036658523676 1468036658523676
changetime__1474198049187875 1474198049187875
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset 4ef41e6fc52a8cef4987d8f513e5aa3d952555cb
```
#!CommitTicketReference repository="" revision="4ef41e6fc52a8cef4987d8f513e5aa3d952555cb"
Merge pull request #302 from twisted/8574-scop-opens

Author: scop
Reviewer: markrwilliams, rodrigc
Fixes: #8574

Handle more file closing with "with"
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
